src/data-access/facades/repository-facade.ts

```diff
@@ -48,21 +48,30 @@ import { IngestedPart } from '../../model/entities/ingested-part'
 import { IngestedSegment } from '../../model/entities/ingested-segment'
 import { CachedSegmentRepository } from '../repositories/cache/cached-segment-repository'
 import { CachedPartRepository } from '../repositories/cache/cached-part-repository'
+import { ConsoleLogger } from '../../console-logger'
 
 export class RepositoryFacade {
   public static createRundownRepository(): RundownRepository {
     const mongoRundownRepository: RundownRepository = new MongoRundownRepository(
-      MongoDatabase.getInstance(),
-      new MongoEntityConverter(),
+      RepositoryFacade.getMongoDatabaseInstance(),
+      RepositoryFacade.createMongoEntityConverter(),
       RepositoryFacade.createSegmentRepository(),
       RepositoryFacade.createPieceRepository()
     )
-    return CachedRundownRepository.getInstance(mongoRundownRepository)
+    return CachedRundownRepository.getInstance(mongoRundownRepository, ConsoleLogger.getInstance())
+  }
+
+  private static createMongoEntityConverter(): MongoEntityConverter {
+    return new MongoEntityConverter(ConsoleLogger.getInstance())
+  }
+
+  private static getMongoDatabaseInstance(): MongoDatabase {
+    return MongoDatabase.getInstance(ConsoleLogger.getInstance())
   }
```

Comment:
Are them being private facade method that bad?

It will mean that for use to add additional arguements to the `MongoEntityConverter` and `MongoDatabase`, we will only need to make the change ones instead of multiple times.

Reply:
It feels wrong to have these as methods. Also, it would be nice to limit the usage of static as much as possible.

Reply:
Thinking abit about it, i more or less just applied the same thing you said regarding creating a facade for creating a logger, just for the creation of a `MongoEntityConverter` and a `MongoDatabase` internally in the facade class.

If the signature for the constructor of those classes change, as they did in the case for this PR, then it will now only be one place to edit, instead of a number of places.

Reply:
Okay that counters my previous argument neatly.

Then i'd like to say that having them reduces duplication, and therefore makes it easier to update, should the class being created need new argument. This was the case for these. I have since then removed reasoning for one of them (MongoEntityConverter) to have this helper method, and have locally mvoed the creation of those back out.

There are at the moment 21 lines in the `RepositoryFacade` that call for a `MongoDatabase`. Without this change, should the `MongoDatabase` need an additional argument, as it did now, then it would require 21 lines to be updated, instead of one. Is this amount of duplication really something you'd rather want?

Regarding the `createExpressErrorHandler` in the `ControllerFacade`, the same thought goes, that it lowers duplication. With this one i even knew that there are two parallel PR which both adds arguments to the creation of `ExpressErrorHandler`, That is currently 5 places that would have had the exact same code snippet, which has been lowered down to 1, that would need changes, should the `ExpressErrorHandler` call for more arguments.

Had there only been 2-3 identical places, i'd not argue it would have been as usefull, as in these remaining instances.